config/crds/v1/bases/elasticsearch.k8s.elastic.co_elasticsearches.yaml

```diff
@@ -534,6 +528,10 @@ spec:
                         description: |-
                           ElasticsearchRefs is a reference to a list of monitoring Elasticsearch clusters running in the same Kubernetes cluster.
                           Due to existing limitations, only a single Elasticsearch cluster is currently supported.
+
+                          This is not optional but was treated as such in previous version of kubernetes-sigs/controller-tools as we also specify
+                          `omitempty` here. See https://github.com/elastic/cloud-on-k8s/pull/8020 and https://github.com/kubernetes-sigs/controller-tools/issues/943
+                          for more details.
```

Comment:
I don't think we want that in the public API docs. I think there is a way to make it not show up there.

Reply:
I'll investigate and update, thanks...

Reply:
Well, I've partially solved this. Adding the `---` removes them from the rendered yaml, but unfortunately I don't think that the crd-ref-docs tool supports this, so it's in the rendered `api-docs` file....

Reply:
Why don't you put the comment above the monitoring struct, basically free-floating so that it doesn not get included. We can then in a second step modify the `crd-ref-docs` tool to follow the k8s convention of ignoring all comments after `---`

Reply:
@pebrc this has been done, and the changes look correct now. The previous full e2e test run: https://buildkite.com/elastic/cloud-on-k8s-operator/builds/9053 only had 2 failures, and they appear to be associated with the recent changes you fixed in #8021.

Reply:
I also created https://github.com/elastic/crd-ref-docs/issues/93